Digitisers/SimHitMerge/src/SimHitMergeAlg.cpp

```diff
@@ -81,13 +85,43 @@ StatusCode SimHitMergeAlg::execute()
               }
               else id_vconb_map[id] = tmp_vconb;
                   
+              if(sanity_check){
+                  if ( test_id_hits_map.find(id) != test_id_hits_map.end()) test_id_hits_map[id].push_back(Simhit);
```

Comment:
I think in this part, `test_id_hits_map[id].push_back(Simhit);` should be work for both case. You can have a check.

Reply:
Sorry, I did not get you point.

Reply:
Even though the `id` does not exist in the map, you can still use `test_id_hits_map[id]`. It will create the empty vector by default.

Reply:
yes, but when id does not exist in the map, I need create a std::vector<edm4hep::SimCalorimeterHit> tmp_vec  first, then I can put the vector in the map. So the judgement is needed. Isn't it?